Taskfile.yml

```
@@ -174,6 +191,36 @@ tasks:
       vars:
         COMPONENT: "{{.TASK}}"
 
+  log-viewer-webui:
+    deps:
+      - "init"
+      - "log-viewer-webui-node-modules"
+      - task: "utils:validate-checksum"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          DATA_DIR: "{{.OUTPUT_DIR}}"
+    dir: "components/log-viewer-webui"
+    vars:
+      CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK}}.md5"
+      OUTPUT_DIR: "{{.G_LOG_VIEWER_WEBUI_BUILD_DIR}}"
+    cmds:
+      - "rm -rf '{{.OUTPUT_DIR}}'"
+      - "rsync -a client server {{.OUTPUT_DIR}}/"
+      - |-
+        cd "{{.OUTPUT_DIR}}/client"
+        PATH="{{.G_LOG_VIEWER_WEBUI_BIN_DIR}}":$PATH npm run build
```

1. We should build the client first and copy only the generated assets to the clp-package. 
2. We should avoid copying the .env file in case in the future it may contain sensitive information for debugging purposes.

e.g., (not tested)
```suggestion
      - |-
        cd client
        PATH="{{.G_LOG_VIEWER_WEBUI_BIN_DIR}}":$PATH npm clean-install
        PATH="{{.G_LOG_VIEWER_WEBUI_BIN_DIR}}":$PATH npm run build
        rsync -a dist "{{.OUTPUT_DIR}}/client/"
      - |-
        cd server
        rsync -a src package-lock.json package.json {{.OUTPUT_DIR}}/server/
```

---

in my experiment, I have noticed that building the client first in the repo has cause pycharm and vscode to get confused on the newly generated files.

This step only copies the files to under build/ (instead of build/clp_package), so it does not violate "copy only the generated assets to the clp-package.", so I would still prefer my original approach. I can update it to not copy .env file

---

Right, sorry for misreading that. It makes sense to copy the sources to the clp/build directory first. We just need to avoid packaging the .env file then.